### PR TITLE
bug fixed

### PR DESCRIPTION
**Pull request by asariev**

### CODE REVIEW

File: docs/api.yml [hunk 1]
```diff
@@ -80,6 +80,15 @@ paths:
     get:
       tags:
         - "User"
+      parameters:
+        - in: query
+          name: size
+          type: string
```
Comment (nursultanturdaliev) [inline docs/api.yml:86]: Type string emes integer bolush kerek emespi?

File: docs/api.yml [hunk 2]
```diff
@@ -80,6 +80,15 @@ paths:
...
+          required: true
+        - in: query
+          name: page
+          type: string
```
Comment (nursultanturdaliev) [inline docs/api.yml:90]: This one too

File: docs/api.yml [hunk 3]
```diff
@@ -92,6 +101,38 @@ paths:
...
+          schema:
+            type: object
+  /api/user/search:
+    parameters:
+      - in: query
+        name: text
+        type: string
+        required: true
+      - in: query
+        name: enabled
```
Comment (nursultanturdaliev) [inline docs/api.yml:125]: enabled=false bolso emne bolot?

File: docs/api.yml [hunk 4]
```diff
@@ -92,6 +101,38 @@ paths:
...
+        type: boolean
+        required: true
+    get:
+      tags:
+        - "User"
+      responses:
+        '200':
+          description: "Successfull fetch"
```
Comment (nursultanturdaliev) [inline docs/api.yml:133]: Description'dy ozgortolu ar bir api'ge jarasha. kandai deisin?

File: docs/api.yml [hunk 5]
```diff
@@ -301,6 +342,33 @@ paths:
...
+                type: string
+              licensePlate:
+                type: string
+              licensePlateState:
+                enum:
+                  - IL
+              make:
+                type: string
+              year:
+                type: string
```
Comment (nursultanturdaliev) [inline docs/api.yml:365]: year is string?

File: src/main/java/io/onroad/eventconsumer/domain/mapper/UserViewMapper.java [hunk 6]
```diff
@@ -31,7 +38,22 @@ public UserView toUserView(User user) {
       userView.setCompany(companyObject);
     }
 
+    if(!user.getRoles().isEmpty()){
+      Set<String> authorities = new HashSet<>();
+      for(Role role: user.getRoles()){
+        authorities.add(role.getName());
+      }
+      userView.setAuthorities(authorities);
+    }
```
Comment (nursultanturdaliev) [inline src/main/java/io/onroad/eventconsumer/domain/mapper/UserViewMapper.java:47]: Let's create a separate method

File: src/main/java/io/onroad/eventconsumer/domain/model/User.java [hunk 7]
```diff
@@ -24,7 +24,11 @@
 import org.springframework.security.core.userdetails.UserDetails;
 
 @Entity
-@Table(name = "users")
+@Table(name = "users",
+        indexes = {@Index(name = "search_drivers", columnList = "firstName, lastName, email, enabled")})
+@NamedQuery(name = "User.findByFirstNameOrLastNameOrEmail",
+        query = "select a from User a where (a.email like CONCAT('%',?1,'%') or a.firstName like CONCAT('%',?1,'%')" +
```
Comment (nursultanturdaliev) [inline src/main/java/io/onroad/eventconsumer/domain/model/User.java:30]: a emes `u` koldonolubu? to est `user` din kyskartmasy.

File: src/main/java/io/onroad/eventconsumer/service/implementation/VehicleServiceImpl.java [hunk 8]
```diff
@@ -29,14 +30,18 @@
...
     @Override
     public VehicleView create(User loggedUser, CreateVehicleRequest createVehicleRequest) {
         createVehicleRequestValidator.validate(loggedUser, createVehicleRequest);
-        if(createVehicleRequest.getCompanyId() == null){
-            createVehicleRequest.setCompanyId(loggedUser.getCompany().getId());
+        Vehicle vehicle = vehicleRepo.save(vehicleEditMapper.create(createVehicleRequest, loggedUser));
+        if(createVehicleRequest.getUserId() != null && createVehicleRequest.getUserId() != 0){
+            Optional<User> user = userRepo.findById(createVehicleRequest.getUserId());
+            user.ifPresent(value -> value.setVehicle(vehicle));
+            userRepo.save(user.get());
```
Comment (nursultanturdaliev) [inline src/main/java/io/onroad/eventconsumer/service/implementation/VehicleServiceImpl.java:42]: Eki bashka company dagy biri birinin userlerine vehicle create kyla berebi oshondo?

**Review by nursultanturdaliev — CHANGES_REQUESTED** [review]

File: docs/api.yml [hunk 9]
```diff
@@ -92,6 +101,38 @@ paths:
...
+      - in: path
+        name: id
+        type: integer
+        required: true
+    delete:
+      tags:
+        - "User"
+      responses:
+        '200':
+          description: "Successful fetch"
```
[the remark this answers is not in the document]
Comment (asariev) [reply docs/api.yml]: updated

File: docs/api.yml [hunk 10]
```diff
@@ -92,6 +101,38 @@ paths:
...
+    parameters:
+      - in: path
+        name: id
+        type: integer
+        required: true
+    delete:
+      tags:
+        - "User"
+      responses:
+        '200':
```
[the remark this answers is not in the document]
Comment (asariev) [reply docs/api.yml]: udpated

File: docs/api.yml (hunk 1 above)
Comment (asariev) [reply docs/api.yml]: updated

File: docs/api.yml (hunk 2 above)
Comment (asariev) [reply docs/api.yml]: updated

File: docs/api.yml (hunk 3 above)
Comment (asariev) [reply docs/api.yml:125]: inactive bolgondordu search kylat

File: docs/api.yml (hunk 4 above)
Comment (asariev) [reply docs/api.yml]: ooba copy paste kylganda udpate kylbapmyn

File: docs/api.yml (hunk 5 above)
Comment (asariev) [reply docs/api.yml]: updated

File: src/main/java/io/onroad/eventconsumer/domain/mapper/UserViewMapper.java (hunk 6 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/domain/mapper/UserViewMapper.java]: udpated

File: src/main/java/io/onroad/eventconsumer/domain/model/User.java (hunk 7 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/domain/model/User.java]: udpated

File: src/main/java/io/onroad/eventconsumer/service/implementation/VehicleServiceImpl.java (hunk 8 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/service/implementation/VehicleServiceImpl.java]: added additional verification that user and logged user is same company dep, user assign kylgan jeri when you create a vehicle, userId will come from dropdown, in dropdown has only same company drivers

**Review by nursultanturdaliev — APPROVED** [review]

**Pull request merged**